services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/SqsBatchFunctions.java

```diff
@@ -0,0 +1,323 @@
+    }
+
+    public static BatchKeyMapper<ChangeMessageVisibilityRequest> changeVisibilityBatchKeyMapper() {
+        return request -> request.overrideConfiguration()
+                                 .map(overrideConfig -> request.queueUrl() + overrideConfig.hashCode())
+                                 .orElse(request.queueUrl());
+    }
+
+    private static SendMessageBatchRequestEntry createSendMessageBatchRequestEntry(String id, SendMessageRequest request) {
+        return SendMessageBatchRequestEntry.builder()
+                                           .id(id)
+                                           .messageBody(request.messageBody())
+                                           .delaySeconds(request.delaySeconds())
+                                           .messageAttributes(request.messageAttributes())
+                                           .messageDeduplicationId(request.messageDeduplicationId())
+                                           .messageGroupId(request.messageGroupId())
+                                           .messageSystemAttributes(request.messageSystemAttributes())
+                                           .build();
+    }
+
+    private static SendMessageResponse createSendMessageResponse(SendMessageBatchResultEntry successfulEntry,
+                                                                 SendMessageBatchResponse batchResponse) {
+        SendMessageResponse.Builder builder = SendMessageResponse.builder()
+                                                                 .md5OfMessageAttributes(successfulEntry.md5OfMessageAttributes())
+                                                                 .md5OfMessageBody(successfulEntry.md5OfMessageBody())
+                                                                 .md5OfMessageSystemAttributes(
+                                                                     successfulEntry.md5OfMessageSystemAttributes())
+                                                                 .messageId(successfulEntry.messageId())
+                                                                 .sequenceNumber(successfulEntry.sequenceNumber());
+        if (batchResponse.responseMetadata() != null) {
+            builder.responseMetadata(batchResponse.responseMetadata());
+        }
+
+        if (batchResponse.sdkHttpResponse() != null) {
+            builder.sdkHttpResponse(batchResponse.sdkHttpResponse());
+        }
+        return builder.build();
+    }
+
+    private static SendMessageResponse createSendMessageResponse(BatchResultErrorEntry failedEntry,
+                                                                 SendMessageBatchResponse batchResponse) {
+        String messageBody = String.format("%s: %s", failedEntry.code(), failedEntry.message());
+        SendMessageResponse.Builder builder = SendMessageResponse.builder()
+                                                                 .md5OfMessageBody(computeMd5Hash(messageBody));
```

Comment:
Hmm, I don't think we should create a message response by ourselves if it fails

Reply:
Yeah that makes sense. Will probably have to refactor core like mentioned above.

Reply:
Yeah, can we return an empty sendMessageResponse for now with a todo?

Reply:
Returning empty sendMessageResponse for now.